lib/ansible/plugins/connection/ssh.py

```diff
@@ -264,7 +264,30 @@
           cli:
             - name: private_key_file
               option: '--private-key'
-
+      private_key:
+          description:
+            - private key contents in PEM format. Requires the SSH_AGENT configuration to be enabled.
+          type: string
+          ini:
+            - section: defaults
+              key: private_key
+          env:
+            - name: ANSIBLE_PRIVATE_KEY
+          vars:
+            - name: ansible_private_key
```

Comment:
add ``ansible_ssh_private_key`` options with higher precedence than the generic, this way we can reuse for other plugins and still have 'plugin specific' option to override or distinguish